### ISSUE TITLE
charAt LowerBound Unknown

### ISSUE THREAD

**Issue by MrigankArora** [typetools/checker-framework#3195]
`charAt` should return a non-negative char. Consider the following test case:

```java
class CharAtTest {
    @NonNegative char test1(@MinLen(1) String s) {
        return s.charAt(0);
    }
}
```

Output:
```
CharAtTest.java:6: error: [return.type.incompatible] incompatible types in return.
        return s.charAt(0);
                       ^
  type of expression: @LowerBoundUnknown char
  method return type: @NonNegative char
1 error
```
Expected: No errors
I'll be working on fixing this issue.

**Comment by kelloggm** [comment]
Related to #2638 and kelloggm#192. If you fix this by editing the annotated JDK, please double-check all the methods in `String.java` that return a `char` or `Character` are annotated as `@NonNegative`

**Comment by mernst** [comment]
> `charAt` should return a non-negative char.

More precisely, in Java `char` is an unsigned type.  `charAt` just returns a `char`, but when that `char` is interpreted as an `int`, it should be a non-negative `int`.

**Comment by mernst** [comment]
The discussion in https://github.com/typetools/checker-framework/pull/3219 suggests that maybe there is a bug related to stub parsing.  We aren't sure, though.
